### PR TITLE
v1.8 backports 2020-07-15

### PR DESCRIPTION
**Pull request by brb**
v1.8 backports 2020-07-15

 * #12443 -- bpf: clear getpeername{4,6} progs (@tklauser)
 * #12415 -- Replace bpf bitfields with a single u8 field and masks (@kkourt)
 * #12427 -- Improve etcd fail-over scenarios (@tgraf)
 * #12456 -- bpf: Fix BPF masq when running with non-hybrid DSR (@brb)
 * #12289 -- Host Policies can break kube if applied incorrectly.  (@jedsalazar)
 * #12440 -- Fix small CRD issue with toGroups (@lbernail)
 * #12457 -- refactor stringSlice type CLI arguments (@JieJhih)
 * #12446 -- cilium: hostport service map fixes (@borkmann)
 * #12451 -- operator: rate limit GC of security identities (@aanm)
 * #12473 -- cilium: simplify hostport's genServiceMappings (@borkmann)
 * #12191 -- Include BPF FS mount status in sysdump output (@soumynathan)
 * #12470 -- pkg/k8s: use copy of objectmeta when fetching from local stores (@aanm)
 * #12462 -- test: Simplify DNS proxy port / TFTP source port conflict test. (@jrajahalme)
 * #12461 -- datapath/linux: protect against concurrent access in NodeValidateImplementation (@tklauser)
 * #12479 -- docs: Improve github actions release docs (@joestringer)
 * #12474 -- policy/api: Support unmanaged entity in policies (@pchaigno)
 * #12492 -- cilium: hostport follow-up (@borkmann)
 * #12494 -- docs: document hostport accessibility in kube-proxy-free guide (@borkmann)
 * #12480 -- Makefile: Fix render-docs with parallel make jobs (@joestringer)
 * #12481 -- Document kvstore leases, update links (@joestringer)
 * #12153 -- doc: Fix clustermesh documentation to set the correct identityMode (@soumynathan)
 * #12497 -- policy: Fix enforcement status of host when PolicyEnforcement=always (@pchaigno)
 * #12495 -- daemon: Fatal on incompatible host firewall options (@pchaigno)
 * #12245 -- Adds TRACE_TO_NETWORK obs label and trace pkts in to-netdev prog. (@Weil0ng)
 * #12486 -- helm: update notes to include Hubble Relay (@Rolinh)
 * #12526 -- clustermesh: Tidy up services connection on failure to reconnect (@joestringer)
 * #12528 -- hubble: add missing code documentation (@Rolinh)
 * #12517 -- Adds documentation for limiting identity-relevant labels used when evaluating Cilium Identities (@seanmwinn)
 * #12529 -- bpf: explicitly set ttl in tunnel key (@borkmann)
 * #12453 -- Add heartbeat to etcd quorum check (@tgraf)
 * #12532 -- docs(troubleshooting):  Remove bugtool related step with --serve flag (@sayboras)
 * #12524 -- endpoint: Write headerfile under full regeneration (@christarazi)
 * #12559 -- bpf: Fix monitor aggregation for 'from-network' (@joestringer)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 12443 12415 12427 12456 12289 12440 12457 12446 12451 12473 12191 12470 12462 12461 12479 12474 12492 12494 12480 12481 12153 12497 12495 12245 12486 12526 12528 12517 12529 12453 12532 12524 12559; do contrib/backporting/set-labels.py $pr done 1.8; done
```

### CODE REVIEW

**Comment by brb** [comment]
test-backport-1.8

**Review by pchaigno — APPROVED** [review]
:+1: for my changes. I also checked changes from @Weil0ng's PR are okay.

**Review by borkmann — APPROVED** [review]
for my PRs

**Review by christarazi — APPROVED** [review]
Looks good for my change

**Review by joestringer — APPROVED** [review]
Backports of my changes LGTM

**Comment by borkmann** [comment]
test-backport-1.8

**Review by rolinh — APPROVED** [review]

**Comment by christarazi** [comment]
test-backport-1.8

File: Documentation/install/upgrade.rst [hunk 1]
```diff
@@ -598,6 +600,7 @@ The following metrics have been renamed:
 * ``cilium_operator_eni_resync_total`` to ``cilium_operator_ipam_resync_total``
 * ``cilium_operator_eni_aws_api_duration_seconds`` to ``cilium_operator_ipam_api_duration_seconds``
 * ``cilium_operator_eni_ec2_rate_limit_duration_seconds`` to ``cilium_operator_ipam_api_rate_limit_duration_seconds``
+* ``cilium_operator_ipam_ec2_resync`` to ``cilium_operator_ipam_resync``
```
Comment (qmonnet) [inline Documentation/install/upgrade.rst]: I think this line sneaked in by mistake in your backport for #12453. The rename comes from  #12413 which has not been backported as far as I can tell.

**Review by qmonnet — CHANGES_REQUESTED** [review]
All backports that have not been reviewed by their authors yet look straightforward with no conflict (or no change at all from what I could see).

I found a single exception below, please fix.

File: Documentation/install/upgrade.rst (hunk 1 above)
Comment (brb) [reply Documentation/install/upgrade.rst]: Good catch! Fixed.

**Comment by brb** [comment]
test-backport-1.8

**Comment by joestringer** [comment]
Hit also #12555 in the k8s-1.13 run.

**Comment by joestringer** [comment]
test-missed-k8s

**Comment by joestringer** [comment]
I believe once we pick up the backport of https://github.com/cilium/cilium/pull/12559, the remaining k8s CI test should begin passing again.

**Comment by borkmann** [comment]
(all green except `Cilium-Ginkgo-Test-k8s`/`Cilium-PR-Ginkgo-Tests-K8s`: https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-K8s/3313/)

**Comment by brb** [comment]
test-backport-1.8

**Comment by brb** [comment]
@joestringer I've included #12559 into this backport.

**Review by qmonnet — APPROVED** [review]
Looks all good to me, let's wait for the tests and we're good to go!

**Comment by borkmann** [comment]
test-missed-k8s

**Comment by borkmann** [comment]
retest-missed-k8s

**Review by tklauser — APPROVED** [review]
👍 for my changes

**Comment by brb** [comment]
Pulling images has failed. E.g.
```
	   Warning  Failed     7m39s                   kubelet, k8s1      Failed to pull image "docker.io/cilium/cilium:v1.7": rpc error: code = Unknown desc = error pulling image configuration: Get https://production.cloudflare.docker.com/registry-v2/docker/registry/v2/blobs/sha256/99/99fc355af08e6fe2381193f228684260b21c0713e46fc4b2510b78cf09e2fe3d/data?verify=1595023904-BW1MMQaahRs0Bh431uW4W5GuIgI%3D: dial tcp: lookup production.cloudflare.docker.com on 8.8.4.4:53: read udp 10.0.2.15:51846->8.8.4.4:53: i/o timeout
```

**Comment by brb** [comment]
retest-missed-k8s

**Comment by brb** [comment]
test-missed-k8s

**Comment by brb** [comment]
test-missed-k8s

**Review by aanm — APPROVED** [review]
LGTM for my changes

**Comment by christarazi** [comment]
test-missed-k8s

**Comment by brb** [comment]
test-missed-k8s

**Pull request merged**